src/PreCICE.jl

```diff
@@ -1332,5 +1340,159 @@ function actionReadIterationCheckpoint()
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 2D problem with 2 vertices:
+
+```julia
+data_id = 1
+valueIndices = [1,2,3]
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0]
+writeBlockVectorGradientData(data_id, valueIndices, gradientValue)
+```
+
+"""
+function writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+    _size, dimensions = size(gradientValues)
+    @assert dimensions == getDimensions()*getDimensions() "Dimensions of vector data in write_block_vector_gradient_data does not match with dimensions in problem definition. Provided dimensions: $dimensions, expected dimensions: $(getDimensions()*getDimensions())"
+
+    ccall((:precicec_writeBlockVectorGradientData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cint}, Ref{Cdouble}), dataID, _size, valueIndices, gradientValues)
+end
+
+@doc """
+
+    writeScalarGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+
+Write gradient data of a scalar data, the value of a specified vertex to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndex::Integer`: Indice of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a scalar data for a 3D problem with 5 vertices:
+
+```julia
+data_id = 1
+vertex_id = 5
+gradientValues = [1.0 2.0 3.0]
+writeScalarGradientData(data_id, vertex_id, gradientValue)
+```
+
+"""
+function writeScalarGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+    _size, dimensions = size(gradientValues)
+    @assert dimensions == getDimensions() "Dimensions of vector data in write_scalar_gradient_data does not match with dimensions in problem definition. Provided dimensions: $dimensions, expected dimensions: $(getDimensions())"
+   
+    ccall((:precicec_writeScalarGradientData, "libprecice"), Cvoid, (Cint, Cint, Ref{Cdouble}), dataID, valueIndex, gradientValues)
+end
+
+
+@doc """
+
+    writeVectorGradientData(dataID::Integer, valueIndex::Integer, gradientValues::AbstractArray{Float64})
+
+Write gradient data of a vector data, the value of a specified vertex to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndex::Integer`: Indice of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 3D problem with 5 vertices:
+
+```julia
+data_id = 1
+vertex_id = 5
+gradientValues = [1.0 2.0 3.0 4.0 5.0 6.0 1.0 2.0 3.0]
```

Comment:
Split this up into the different dimensions?

Reply:
it is a single vector/[1x9] matrix there is no need to splitting

Reply:
Now I understand, maybe a comment on how the vector needs to be passed would be helpful? e.g. in the [python bindings](https://github.com/kursatyurt/python-bindings/blob/66e9339e0e1f1bbc22559358236b6cf3111d2511/cyprecice/cyprecice.pyx#L1278-L1282) there is a comment on the format the matrix needs to be passed.